app/main.py

```diff
@@ -1,34 +1,134 @@
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
         # Create decks and save them to a list `self.decks`
-        pass
+        self.decks = self.__create_decks(start, end)
+        self.is_drowned = is_drowned
+
+    @staticmethod
+    def __create_decks(start: tuple, end: tuple) -> list[Deck]:
+        deck_coordinates = []
+        common_coordinate = "row" if start[0] in end else "column"
+        while start != end:
+            deck_coordinates.append((start[0], start[1]))
+            if common_coordinate == "row":
+                start = (start[0], start[1] + 1)
+            else:
+                start = (start[0] + 1, start[1])
+        deck_coordinates.append((start[0], start[1]))
+        deck_instances = [
+            Deck(row, column) for row, column in deck_coordinates
+        ]
+        return deck_instances
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.coordinates == (row, column):
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        if not any(deck.is_alive for deck in self.decks):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: List[tuple]) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.ships = ships
+        self.field = self.__create_ships()
+        self.__validate_field()
 
-    def fire(self, location: tuple):
+    def __create_ships(self) -> dict:
+        deck_and_its_ship = {}
+        ships = [Ship(start, end) for start, end in self.ships]
+        for ship in ships:
+            for deck in ship.decks:
+                deck_and_its_ship[deck.coordinates] = ship
+        return deck_and_its_ship
+
+    def fire(self, location: tuple) -> str:
         # This function should check whether the location
         # is a key in the `self.field`
         # If it is, then it should check if this cell is the last alive
         # in the ship or not.
-        pass
+        if location in self.field:
+            ship = self.field.get(location)
+            ship.fire(location[0], location[1])
+            if any(deck.is_alive for deck in ship.decks):
+                return "Hit!"
+            return "Sunk!"
+        return "Miss!"
+
+    def __create_field(self) -> list:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]